lib/web3/eth/patches.rb

```diff
@@ -0,0 +1,42 @@
+unless Hash.method_defined?(:compact)
```

Comment:
These look like they're all also in `activesupport`. Alternately, you could cherry pick require the bits of `activesupport` that you want.

http://guides.rubyonrails.org/active_support_core_extensions.html#cherry-picking-a-definition